### ISSUE TITLE
[parallel] file openwith/openps does not work with no filename specified

### ISSUE THREAD

**Issue by rderooy** [joncampbell123/dosbox-x#3166]
### What version(s) of DOSBox-X have this bug?

0.83.21git

### Describe the bug

When printing to a file and using openwith: or openps: the file never gets opened.
openpcl: and openerror: almost certainly has the same issue, but was not tested.

Here is the test case config:

```
[dosbox]
captures=.

[parallel]
parallel1=file timeout:1000 openwith:xdg-open
```

Starting DOSBox-X with this config and on the prompt doing a ``DIR > LPT1:`` will result in an error dialog popping up stating that the "The requested file printing handler failed to complete.".

In addition the console shows this error:
```
LOG: SHELL:Redirect output to LPT1
LOG: Capturing Parallel Port Stream to /home/rderooy/command_000.prt
LOG: Parallel 1: File closed.
   xdg-open - opens a file or URL in the user's preferred
   application

Synopsis

   xdg-open { file | URL }

   xdg-open { --help | --manual | --version }

Use 'man xdg-open' or 'xdg-open --manual' for additional info.
```
Basically it appears that xdg-open was called without arguments.

If I change the line as such, it does work, but then new print jobs will overwrite the old once.
```
parallel1=file timeout:1000 file:output.prn openwith:xdg-open
```

**Comment by Wengier** [comment]
I could reproduce this issue, which happens only if a file name is not provided. Please check out the latest code, which hopefully fixes this issue.

**Comment by rderooy** [comment]
I tried the latest code, but it made no difference. I still get the same errors.

**Comment by Wengier** [comment]
It looks like the previous fix only worked for Windows. Please try the updated code.
